Sources/PackageModel/UserToolchain.swift

```diff
@@ -410,23 +410,23 @@ public final class UserToolchain: Toolchain {
     static func deriveMacOSSpecificSwiftTestingFlags(
         derivedSwiftCompiler: AbsolutePath,
         fileSystem: any FileSystem
-    ) -> [String] {
+    ) -> (swiftCFlags: [String], linkerFlags: [String]) {
         // If this is CommandLineTools all we need to add is a frameworks path.
         if let frameworksPath = try? AbsolutePath(
             validating: "../../Library/Developer/Frameworks",
             relativeTo: resolveSymlinks(derivedSwiftCompiler).parentDirectory
         ), fileSystem.exists(frameworksPath.appending("Testing.framework")) {
-            return [
-                "-F", frameworksPath.pathString,
-                "-Xlinker", "-rpath",
-                "-Xlinker", frameworksPath.pathString
-            ]
+            return (swiftCFlags: [
+                "-F", frameworksPath.pathString
```

Comment:
Is this framework search path not also needed for the linker?

Reply:
This appears to be sufficient, I tested with commandline tools only and -F + `-Xlinker -rpath` builds and runs the swift-testing tests without issues.